### PR TITLE
Lodash lang

### PR DESCRIPTION
**Pull request by striok**

### CODE REVIEW

File: app/datasets.js [hunk 1]
```diff
@@ -0,0 +1,225 @@
+'use strict';
+
+module.exports = {
+    castArray: function ()
+    {
+        return [];
```
Comment (Nigatsu) [inline app/datasets.js:6]: Pass when I copy and paste answer from test file

File: app/datasets.js [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+    },
+    clone: function ()
+    {
+        return [];
```
Comment (Nigatsu) [inline app/datasets.js:10]: Pass when I copy and paste answer from test file

File: app/datasets.js [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+    },
+    cloneDeep: function (obj)
+    {
+        return [];
```
Comment (Nigatsu) [inline app/datasets.js:14]: Only I need to do is copy _obj_ to array to pass...

File: app/datasets.js [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+     return [];
+     },
+     cloneWith : function(){
+     return [];
+     },
+     */
+
+    eq: function ()
+    {
+        return [];
```
Comment (Nigatsu) [inline app/datasets.js:18]: I enter two empty objects as a guess and it pass, but why?

File: app/datasets.js [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+    },
+    gte: function ()
+    {
+        return [];
+    },
+    isArguments: function ()
+    {
+        return [];
```
Comment (Nigatsu) [inline app/datasets.js:30]: Is not enough information to achieve that, even in lodash docs

File: app/datasets.js [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+    },
+
+    /*
+     cloneDeepWith :function(){
+     return [];
+     },
+     cloneWith : function(){
+     return [];
+     },
+     */
```
Comment (Nigatsu) [inline app/datasets.js]: What is it for?

File: app/datasets.js [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+    isDate: function ()
+    {
+        return [];
+    },
+
+    /*
+     isElement: function(){
+     return [document.body];
+     },
+     */
```
Comment (Nigatsu) [inline app/datasets.js]: What is it for?

File: app/datasets.js [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+    {
+        return [];
+    },
+    isEqual: function ()
+    {
+        return [];
+    },
+    isEqualWith: function ()
+    {
+        return [];
```
Comment (Nigatsu) [inline app/datasets.js:70]: Two empty arrays can solve that

File: app/datasets.js [hunk 9]
```diff
@@ -0,0 +1,225 @@
...
+    },
+    isError: function ()
+    {
+        return [];
+    },
+    isFinite: function ()
+    {
+        return [];
```
Comment (Nigatsu) [inline app/datasets.js:78]: Even example from lodash docs can't solve that.

File: app/datasets.js [hunk 10]
```diff
@@ -0,0 +1,225 @@
...
+    },
+    isNaN: function ()
+    {
+        return [];
+    },
+    isNative: function ()
+    {
+        return [];
```
Comment (Nigatsu) [inline app/datasets.js:110]: Even example from lodash docs can't solve that.

File: test/unit/lodash.spec.js [hunk 11]
```diff
@@ -0,0 +1,1287 @@
...
+         describe.skip('cloneDeepWith', function () {
+
+         // This function freezes Mocha (when you use document.body). In web browser works fine. Perhaps need some DOM?
+         });
+
+         describe.skip('cloneWith', function () {
+
+         // This function freezes Mocha (when you use document.body). In web browser works fine. Perhaps need some DOM?
+         });
+         */
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js]: What is it for?

File: test/unit/lodash.spec.js [hunk 12]
```diff
@@ -0,0 +1,1287 @@
...
+         it('should verify params', function () {
+         expect(params).to.have.length(1);
+         });
+
+         it('should return true if value is a DOM element', function () {
+         expect(_.isElement.apply(_, datasets.isElement(params))).to.eql(true);
+         });
+         });
+
+         */
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js]: What is it for?

File: test/unit/lodash.spec.js [hunk 13]
```diff
@@ -0,0 +1,1287 @@
...
+            before(function ()
+            {
+                params = datasets.isNative();
+                value = params[0];
+            });
+
+            it('should verify params', function ()
+            {
+                expect(params).to.have.length(1);
+                // check name of user method ?
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js]: What is it for?

File: test/unit/lodash.spec.js [hunk 14]
```diff
@@ -0,0 +1,1287 @@
...
+
+            it('should return the converted integer', function ()
+            {
+                expect(_.toNumber.apply(_, datasets.toNumber(params))).to.eql(2016);
+            });
+        });
+
+        describe('toPlainObject', function ()
+        {
+            // User must use _.assign to obtain the effect
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js]: What is it for?

File: app/datasets.js [hunk 15]
```diff
@@ -0,0 +1,225 @@
...
+    },
+    toSafeInteger: function ()
+    {
+        return [];
+    },
+    toString: function ()
+    {
+        return [];
+    }
```
Comment (Nigatsu) [inline app/datasets.js:207]: Many tests pass with answer from tests file

File: app/datasets.js (hunk 9 above)
Comment (striok) [inline app/datasets.js:78]: It should be ok for - _Infinity_ becouse test return false when value isn't finite number.

**Pull request merged**